tests/test_dock_layout.py

```diff
@@ -69,3 +70,131 @@ def test_validation_dock_layout2(enaml_qtbot, enaml_sleep):
+    win.show()
+    
+    wait_for_window_displayed(enaml_qtbot, win)
+    enaml_qtbot.wait(enaml_sleep)
+    _children = win.children[0].children
+    btn_save, btn_restore, btn_add, cmb_styles, cbx_evts, dock = _children
+        
+    # Enable dock events
+    enaml_qtbot.mouseClick(cbx_evts.proxy.widget, Qt.LeftButton)
+    enaml_qtbot.wait(enaml_sleep)
+    
+    with pytest.warns(DockLayoutWarning):
+        # Change styles
+        cmb_styles.proxy.widget.setFocus()
+        enaml_qtbot.keyClick(cmb_styles.proxy.widget, Qt.Key_Up)
+        
+        # Save layout
+        enaml_qtbot.mouseClick(btn_save.proxy.widget, Qt.LeftButton)
+        enaml_qtbot.wait(enaml_sleep)
+        
+        # Toggle closable
+        di = dock.find('item_2')
+        di.closable = False
+        enaml_qtbot.wait(enaml_sleep)
+        di.closable = True
+        
+        # Check alert
+        di.alert('info')
+        
+        # Maximize it
+        tb = di.proxy.widget.titleBarWidget()
+        enaml_qtbot.mouseClick(tb._max_button, Qt.LeftButton)
+        enaml_qtbot.wait(enaml_sleep)
+        
+        # Minimize it
+        enaml_qtbot.mouseClick(tb._restore_button, Qt.LeftButton)
+        enaml_qtbot.wait(enaml_sleep)
+    
+        # Pin it
+        enaml_qtbot.mouseClick(tb._pin_button, Qt.LeftButton)
+        enaml_qtbot.wait(enaml_sleep)
+        
+        # TODO: Open and close it using clicks
+        dock_area = dock.proxy.widget
+        for c in dock_area.dockBarContainers():
+            dock_bar = c[0]
+            dock_area.extendFromDockBar(dock_bar)
+            enaml_qtbot.wait(enaml_sleep)
+            dock_area.retractToDockBar(dock_bar)
+            enaml_qtbot.wait(enaml_sleep)
+        
+        # Unpin
+        enaml_qtbot.mouseClick(tb._pin_button, Qt.LeftButton)
+        enaml_qtbot.wait(enaml_sleep)
+        
+        # Note the location of the dock item title
+        title = tb._title_label
+        ref = win.proxy.widget
+        
+        # Maximize and minimize it by double clicking
+        pos = title.mapTo(tb, title.pos())
+        enaml_qtbot.mouseDClick(tb, Qt.LeftButton, pos=pos)
+        enaml_qtbot.wait(enaml_sleep)
+        enaml_qtbot.mouseDClick(tb, Qt.LeftButton, pos=pos)
+        enaml_qtbot.wait(enaml_sleep)
+        
+        # Float it
+        op = FloatItem(item=di.name)
+        dock.update_layout(op)
+        enaml_qtbot.wait(enaml_sleep)
+        
+        # Link it
+        enaml_qtbot.mouseClick(tb._link_button, Qt.LeftButton)
+        enaml_qtbot.wait(enaml_sleep)
+        
+        # Unlink it
+        enaml_qtbot.mouseClick(tb._link_button, Qt.LeftButton)
+        enaml_qtbot.wait(enaml_sleep)
+        
+        # Restore it
+        op = InsertTab(item=di.name, target='item_1')
+        dock.update_layout(op)
+        enaml_qtbot.wait(enaml_sleep)
+        
+        # Drag it around
+        # TODO: this doesn't currently drag the window. It seems like QTest
```

Comment:
Since this does not work, is it worth keeping it around ?

Reply:
It can be removed, but I was hoping someone had more insight on how to do this.  If you add a long sleep and drag it around manually it catches about another 500 lines of coverage.

Reply:
Ok then it is worth investigating. But some quick searches on google did not point to any solution.